packages/cell-toolbar/src/celltoolbartracker.ts

```diff
@@ -50,7 +50,12 @@ export class CellToolbarTracker implements IDisposable {
     this._toolbar.changed.connect(this._onToolbarChanged, this);
 
     // Only add the toolbar to the notebook's active cell (if any) once it has fully rendered and been revealed.
-    void panel.revealed.then(() => this._onActiveCellChanged(panel.content));
+    void panel.revealed.then(() => {
+      // Wait one frame (at 60 fps) for the panel to render the first cell, then display the cell toolbar on it if possible.
+      setTimeout(() => {
```

Comment:
Did you consider using `requestAnimationFrame` instead?

Reply:
You may also try to wait for `Cell.ready`:

https://github.com/jupyterlab/jupyterlab/blob/98e684cfbe8982b65d9ea518ccb9801762cc8a8a/packages/cells/src/widget.ts#L398

Reply:
I tried replacing the block above with code that listens to the `ready` promise @fcollonval mentioned:

```ts
    void panel.content.activeCell?.ready.then(() => {
      this._onActiveCellChanged(panel.content);
    });
```

I saw the cell toolbar appear overlapping content, as in the "before" screen shot in the description.

Reply:
I also tried replacing the block above with code that calls `requestAnimationFrame` as @krassowski mentioned:

```ts
    void panel.revealed.then(() => {
      // Wait one frame for the panel to render the first cell, then display the cell toolbar on it if possible.
      requestAnimationFrame((timestamp) => {
        this._onActiveCellChanged(panel.content);
      });
    });
```

I also saw the cell toolbar appear as in the "before" screen shot, overlapping cell content.

Reply:
Thanks for testing the suggestion @JasonWeill